### PR TITLE
Remove "using std::string" from port.h.

### PR DESCRIPTION
**Pull request by gregcoombe**
Exposing the "using" directive in the port header will cause it
to be leaked into any codebase that uses this library. While it
is possible to just fix up the headers and use the directive
only in the source files, this change uses std::string
everywhere.

### CODE REVIEW

**Comment by gregcoombe** [comment]
@googlebot I signed it!

File: src/s2/s2builderutil_closed_set_normalizer_test.cc [hunk 1]
```diff
@@ -71,22 +71,22 @@ class NormalizeTest : public testing::Test {
                      DuplicateEdges::KEEP, SiblingPairs::KEEP));
   }
 
-  void Run(const string& input_str, const string& expected_str);
+  void Run(const std::string& input_str, const std::string& expected_str);
```
Comment (jmr) [inline src/s2/s2builderutil_closed_set_normalizer_test.cc:74]: The way `s2geometry` likes to do this in `.cc` files is to do `using std::string;` after the `#include`s, then leave plain `string` in the rest of the file. I can merge it if you do that.

`.h` is good like you have it.

**Comment by gregcoombe** [comment]
Ok, I've updated the CL to use the `using` declarations in the source files, and the `std::` in the headers (despite some personal misgivings about the readability of such an approach).

**Comment by jmr** [comment]
I will close and re-open this in hope of getting travis-ci to notice the .travis.yml changes. The current CMakeLists.txt is incompatible with the old .travis.yml.

File: src/s2/s2shapeutil_edge_iterator.cc [hunk 2]
```diff
@@ -17,9 +17,11 @@
 
 #include "s2/third_party/absl/strings/str_cat.h"
 
+using std::string;
+
 namespace s2shapeutil {
 
-EdgeIterator::EdgeIterator(const S2ShapeIndex* index)
+EdgeIterator::EdgeIterator(const S2ShapeIndex *index)
```
Comment (jmr) [inline src/s2/s2shapeutil_edge_iterator.cc:24]: Did you somehow automatically reformat this? Please don't do that.

File: src/s2/s2text_format.h [hunk 3]
```diff
@@ -263,26 +263,26 @@ std::unique_ptr<MutableS2ShapeIndex> MakeIndex(absl::string_view str);
...
+std::string ToString(const S2CellUnion& cell_union);
+std::string ToString(const S2Loop& loop);
+std::string ToString(S2PointLoopSpan loop);
+std::string ToString(const S2Polyline& polyline);
+std::string ToString(const S2Polygon& polygon, const char* loop_separator = ";\n");
+std::string ToString(const std::vector<S2Point>& points);
+std::string ToString(const std::vector<S2LatLng>& points);
+std::string ToString(const S2LaxPolylineShape& polyline);
+std::string ToString(const S2LaxPolygonShape& polygon,
                 const char* loop_separator = ";\n");
```
Comment (jmr) [inline src/s2/s2text_format.h:279]: indentation

File: src/s2/third_party/absl/strings/str_cat.h [hunk 4]
```diff
@@ -304,27 +304,27 @@ class AlphaNum {
...
-  return string(a.data(), a.size());
+ABSL_MUST_USE_RESULT inline std::string StrCat(const AlphaNum& a) {
+  return std::string(a.data(), a.size());
 }
 
-ABSL_MUST_USE_RESULT string StrCat(const AlphaNum& a, const AlphaNum& b);
-ABSL_MUST_USE_RESULT string StrCat(const AlphaNum& a, const AlphaNum& b,
+ABSL_MUST_USE_RESULT std::string StrCat(const AlphaNum& a, const AlphaNum& b);
+ABSL_MUST_USE_RESULT std::string StrCat(const AlphaNum& a, const AlphaNum& b,
                                    const AlphaNum& c);
```
Comment (jmr) [inline src/s2/third_party/absl/strings/str_cat.h:321]: indentation

Comment (jmr) [reply src/s2/third_party/absl/strings/str_cat.h:321]: other places in this file, too

File: src/s2/third_party/absl/strings/string_view.h [hunk 5]
```diff
@@ -27,18 +27,18 @@
 #ifndef S2_THIRD_PARTY_ABSL_STRINGS_STRING_VIEW_H_
 #define S2_THIRD_PARTY_ABSL_STRINGS_STRING_VIEW_H_
 
-#include <algorithm>
 #include "s2/third_party/absl/base/config.h"
+#include <algorithm>
```
Comment (jmr) [inline src/s2/third_party/absl/strings/string_view.h:31]: don't reorder the includes. more non std::string changes in this file.

**Comment by jmr** [comment]
Tests are failing.

**Comment by jmr** [comment]
Obsoleted by #213 .

**Pull request closed without merging**